### ISSUE TITLE
dashboard settings not working for CAPEX vs OPEX dashboard, and potentially more.

### ISSUE THREAD

**Issue by JIMENOFONSECA** [architecture-building-systems/CityEnergyAnalyst#1870]
I basically open the dashboard and get as default this plot:

![chrome_hcGcPyY0lr](https://user-images.githubusercontent.com/8973186/55846805-9c1fa200-5b79-11e9-840a-2abab480eb71.png)

1. I then open the settings and see several variables that do not have anything to do with the plot
i.e., detailed electricity pricing, multi-criteria, network-type, region.

2. In addition, if I either save changes or change any parameter in the settings, the code brakes.

![chrome_9BMJd4xVCB](https://user-images.githubusercontent.com/8973186/55846870-e43ec480-5b79-11e9-9e91-ba0629427c22.png)

![chrome_Nk3jRVgIc1](https://user-images.githubusercontent.com/8973186/55846874-e7d24b80-5b79-11e9-9cad-4e038a1467c2.png)

![chrome_x9qnMC9jOA](https://user-images.githubusercontent.com/8973186/55846886-ea34a580-5b79-11e9-81fa-cb739b532d2b.png)

![chrome_l6I3qjyPQn](https://user-images.githubusercontent.com/8973186/55846889-ec96ff80-5b79-11e9-8ce2-b4d233a5d852.png)

![chrome_OE2Y0n5A4A](https://user-images.githubusercontent.com/8973186/55846893-ef91f000-5b79-11e9-8442-3434fd81789a.png)

**Comment by JIMENOFONSECA** [comment]
@daren-thomas this behavior also happens in the 'Capacity installed in generation 2 plots'

**Comment by daren-thomas** [comment]
Ah. I think I've noticed this two. I think I'm going to spend this afternoon going after these issues.

**Comment by daren-thomas** [comment]
Some of this is fixed. Other stuff not yet.

- regarding 1: detailed electricity pricing, multi-criteria, network-type, region... for each of the optimization plots, we need to define the variables...

- regarding 2: fixed that, but still I'd like to work some more on that (formatting of BooleanParameter is not as nice as in the tools)

In addition, when setting the generation to 1 (working from cooling-case-workflow output) the multi-criteria files are missing for that generation. This causes an Internal Server Error instead of the expected output (please run tool XY) because the multi-criterea-analysis meta-data hasn't been added yet to schemas.yml.

**Comment by daren-thomas** [comment]
there is a change in #1984 (#2083) to the PlotBase class that needs to be merged before I can continue on this.

**Comment by daren-thomas** [comment]
@reyery how can we get this merged?

**Comment by reyery** [comment]
@daren-thomas I am still getting the same error regarding something about checkpoints. 
```
Traceback (most recent call last):
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask\app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask\app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask\app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\CityEnergyAnalyst\cea\interfaces\dashboard\plots\routes.py", line 109, in route_add_plot_to_dashboard
    dashboard.add_plot(category, plot_id)
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\CityEnergyAnalyst\cea\plots\__init__.py", line 109, in add_plot
    plot = plot_class(self.config.project, parameters, self.cache)
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\CityEnergyAnalyst\cea\plots\optimization\pareto_capacity_installed.py", line 24, in __init__
    super(ParetoCapacityInstalledPlot, self).__init__(project, parameters, cache)
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\CityEnergyAnalyst\cea\plots\optimization\__init__.py", line 63, in __init__
    self.data_address = locating_individuals_in_generation_script(self.generation, self.locator)
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\CityEnergyAnalyst\cea\analysis\multicriteria\optimization_post_processing\locating_individuals_in_generation_script.py", line 58, in locating_individuals_in_generation_script
    data_generation = preprocessing_generations_data(locator, generation)
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\CityEnergyAnalyst\cea\analysis\multicriteria\optimization_post_processing\locating_individuals_in_generation_script.py", line 156, in preprocessing_generations_data
    with open(locator.get_optimization_checkpoint(generations), "rb") as fp:
IOError: [Errno 2] No such file or directory: 'C:\\plot-test\\baseline\\outputs\\data\\optimization\\master\\CheckPoint_3'
```

We could merge #2090 first if you want since you mentioned that it includes other fixes, then I submit another issue regarding the error above?

**Comment by daren-thomas** [comment]
I'm pretty sure that's because the generation is set to 3 by default and you only have data for 2 generations. but also, those plots are all going to be changed by jimeno anyway, so not merging because of obsolete plots isn't a good idea.

instead, test with other plots to see if the parameters are being set / saved properly. the issue was mainly with parameters that had hyphens in them. and also boolean parameters (this PR, if i remember correctly - we have waited waaaay too long to merge, also makes sure they show up nicely with that green toggle thing)
